engine/src/main/java/org/destinationsol/assets/audio/OggMusicManager.java

```diff
@@ -133,4 +144,64 @@ private void stopMusic() {
     public void changeVolume(GameOptions options) {
         currentlyPlaying.setVolume(options.musicVolume.getVolume());
     }
+
+    /**
+     * Registers all music from the module of shipName
+     *
+     * @param shipName Name of the ship that is loaded when creating a new game
+     */
+    public void registerModuleMusic(String shipName) {
+        Set<ResourceUrn> configUrnList = Assets.getAssetHelper().list(Json.class, "[a-zA-Z]*:playerSpawnConfig");
+
+        for (ResourceUrn configUrn : configUrnList) {
+            Json json = Assets.getJson(configUrn.toString());
+            JSONObject rootNode = json.getJsonValue();
+
+            if (rootNode.keySet().contains(shipName) && rootNode.get(shipName) instanceof JSONObject) {
+                JSONObject shipNode = rootNode.getJSONObject(shipName);
+
+                String moduleName = shipNode.getString("hull").split(":")[0];
+                Json musicJson;
+
+                try {
+                    musicJson = Assets.getJson(moduleName + ":musicConfig");
+                } catch (RuntimeException e) {
+                    logger.warn("Music config not found for module " + moduleName + ":musicConfig");
+                    return;
+                }
+
+                MusicConfig musicConfig = MusicConfig.load(moduleName, musicJson.getJsonValue());
+                Map<String, List<String>> musicMap = musicConfig.getMusicMap();
+
+                for (String music : musicMap.get(MENU_MUSIC_SET)) {
+                    registerMusic(MENU_MUSIC_SET, music);
+                }
+                for (String music : musicMap.get(GAME_MUSIC_SET)) {
+                    registerMusic(GAME_MUSIC_SET, music);
+                }
+            }
+
+            json.dispose();
+        }
+    }
+
+    /**
+     * Registers all module menu music
+     */
+    public void registerAllMenuMusic() {
```

Comment:
Actually, the music is only registered once you choose a ship (module) and start the game. Since no module is selected when the game is launched, I thought it would be a better decision to load all module menu music.